Framework/CurveFitting/src/Functions/InelasticDiffRotDiscreteCircle.cpp

```diff
@@ -150,11 +150,11 @@ void InelasticDiffRotDiscreteCircle::setWorkspace(
       break;
     }
 
-    const auto &det = spectrumInfo.detector(idx);
+    const auto detID = spectrumInfo.detector(idx).getID();
```

Comment:
Could you go even further and avoid the detector access and use the `DetectorInfo::getDetectorIDs()` and just the index to get the right detector id directly. Remember not to make a copy of it. You get a const reference to that vector via the public method.